public/scripts/d3/d3.js

```diff
@@ -1 +1,169 @@
-//  Add d3 functions here
+d3.json(echo.setup.jsonUrl, (err, data) => {
+  const startingCx = 160;
+  const startingCy = 120;
+
+  data.sort((a, b) => d3.descending(a.avgRating, b.avgRating));
+
+  var sortedByTag = echo.helpers.binByKey('tag', data);
+  var tagPositions = sortedByTag.map((tagObj, i) =>
+  ({ tag: tagObj.tag, cx: startingCx, cy: startingCy + 200 * i, nodes: tagObj.nodes }));
+
+  var nodeTagsArrayToLinksAndNodes = function(nodes) {
```

Comment:
The position of function declarations is confusing.

```
const SOME_CONSTANT = bla bla
// other const decs

// The all function declarations

// then

var sortedByTag = echo.helpers.binByKey('tag', data);
var tagPositions = sortedByTag.map((tagObj, i) =>
  ({ tag: tagObj.tag, cx: startingCx, cy: startingCy + 200 * i, nodes: tagObj.nodes }));
```

Ie define functions and constants, then write the main code that is executing top level logic.

right now the function definitions are sandwiched between the top level logic